app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -443,8 +443,8 @@ def parse_service_port_config(port_config, service_id)
         :name        => port_config.name,
         :protocol    => port_config.protocol,
         :port        => port_config.port,
-        :target_port => port_config.targetPort,
-        :node_port   => port_config.nodePort,
+        :target_port => (port_config.targetPort unless port_config.targetPort == 0),
```

Comment:
I'm not sure what is the meaning of an empty name.
When I feed k8s with an empty string for target port it returns targetPort == port so I don't think it is a possible value.